### PR TITLE
10Gigabit Ethernet driver for AMD SoC

### PR DESCRIPTION
**Pull request by swhite2**
This patch has the driver for 10Gigabit Ethernet controller in AMD
SoC. This driver is written compatible to the Iflib framework. The
existing driver is for the old version of hardware. The submitted
driver here is for the recent versions of the hardware where the Ethernet
controller is PCI-E based.

Submitted by:	Rajesh Kumar <rajesh1.kumar@amd.com>
MFC after:	1 month
Relnotes:	yes
Differential Revision:	https://reviews.freebsd.org/D25793

### CODE REVIEW

File: sys/conf/files.arm64 [hunk 1]
```diff
@@ -204,11 +204,16 @@ dev/acpica/acpi_if.m		optional	acpi
...
+dev/axgbe/if_axgbe.c		optional	axa
+dev/axgbe/xgbe-desc.c		optional	axa
+dev/axgbe/xgbe-dev.c		optional	axa
+dev/axgbe/xgbe-drv.c		optional	axa
+dev/axgbe/xgbe-mdio.c		optional	axa
+dev/axgbe/xgbe-sysctl.c		optional	axa
+dev/axgbe/xgbe-txrx.c		optional	axa
+dev/axgbe/xgbe_osdep.c		optional	axa
+dev/axgbe/xgbe-phy-v1.c		optional	axa
+>>>>>>> 7113afc84c0b... 10Gigabit Ethernet driver for AMD SoC
```
Comment (fichtner) [inline sys/conf/files.arm64:216]: looks like a merge fail?

Comment (swhite2) [reply sys/conf/files.arm64]: Woops! Fixed now.

**Comment by swhite2** [comment]
Since the patches are not yet upstreamed, I'm committing them now for 20.7 release.

**Comment by fichtner** [comment]
@swhite2 Main commits are cdd4a70cba07ed843ec02ba3e1fb5108512e6a9a + 61c2b482e316fe331f58754b20e4255916f3d213 + 21ec4b0746dda458c7f3bd9bcc70f63791cc39f3 right? I need to clean up the commit history a bit

**Comment by swhite2** [comment]
> @swhite2 Main commits are [cdd4a70](https://github.com/opnsense/src/commit/cdd4a70cba07ed843ec02ba3e1fb5108512e6a9a) + [61c2b48](https://github.com/opnsense/src/commit/61c2b482e316fe331f58754b20e4255916f3d213) + [21ec4b0](https://github.com/opnsense/src/commit/21ec4b0746dda458c7f3bd9bcc70f63791cc39f3) right? I need to clean up the commit history a bit

@fichtner Correct! Although https://github.com/opnsense/src/commit/21ec4b0746dda458c7f3bd9bcc70f63791cc39f3 is still a commit made by me, but the differential is open and shouldn't take long to get upstreamed.

Sorry for cluttering the commit history, but I was unsure of when you were planning to commit this to 20.7, so I made sure all relevant bits were in. Up-to-date description of this PR is here: https://github.com/opnsense/src/issues/93.

**Comment by fichtner** [comment]
Sorry I forgot about #93 :D

**Comment by fichtner** [comment]
Cherry-picked off your commits, all done. Thanks! :)

**Pull request closed without merging**